src/main/java/ro/msg/learning/shop/services/StrategyFactory.java

```diff
@@ -0,0 +1,37 @@
+package ro.msg.learning.shop.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.*;
+import org.springframework.stereotype.Service;
+
+@Service
+@Configuration
+
+public class StrategyFactory {
+
+    private final String strategyFindLocation;
+    private final SingleLocationStrategy singleLocationStrategy;
+    private final MoreAbundantStrategy moreAbundantStrategy;
+
+    @Autowired
+    public StrategyFactory(@Value("${strategy.findLocation:SingleLocationStrategy}") String strategyFindLocation,
```

Comment:
Please create an Enum for the Strategy Type. On application.properties set as a value for strategy.findLocation one of the enum values. And here map that value from the property to strategyFindLocation variable using @Value annotation. Remove singleLocationStrategy and moreAbundantStrategy from this class because here you should create just one of these 2 strategies. Then on getStrategy method, based on the value of strategyFindLocation variable, create and return the corresponding strategy object: SingleLocationStrategy or MoreAbundantStrategy (which should be by the way "**Most**AbundantStrategy").

Reply:
DONE